### PR TITLE
Feature/53 test uses proof property in did document

### LINKED ISSUE #53: [vda-verification-base] Create test that uses proof property in DID document
Create a new unit test in [VDA Verification Base](https://github.com/verida/blockchain-contracts/tree/develop/VDA-Verification-Base) that verifies the `proof` string created by the Verida protocol works correctly with the on chain `proof` verification.

Requirements:

1. Create a new unit test in [VDA Verification Base](https://github.com/verida/blockchain-contracts/tree/develop/VDA-Verification-Base)
2. Create a new DID document with a context, save it using DID Client, then fetched it from the blockchain ([see example in unit test](https://github.com/verida/verida-js/blob/release/2.0-acacia/packages/did-client/test/did-client.tests.ts#L59))
3. Make an on chain verification request (similar to existing [unit test](https://github.com/verida/blockchain-contracts/blob/f6abcc0f9dfa39d35452cef99a48399fc0f9832d/VDA-Verification-Base/test/index.ts#L107)), except:
  - Use the `proof` from the DID document (See PR 243 below) instead of generating your own [proof](https://github.com/verida/blockchain-contracts/blob/f6abcc0f9dfa39d35452cef99a48399fc0f9832d/VDA-Verification-Base/test/index.ts#L123)
  - The `signature` should be generated by signing data using the application context private signing key. This can be obtained by [keyring.getKeys().signPrivateKey](https://github.com/verida/verida-js/blob/68c62dd848d376771f3b15e8472f10bda49ed4cd/packages/did-document/src/did-document.ts#L108), also see [Keyring.getKeys() method](https://github.com/verida/verida-js/blob/68c62dd848d376771f3b15e8472f10bda49ed4cd/packages/keyring/src/keyring.ts#L42)
5. Verify the onchain verification works as expected

Also see:

1. [PR 243](https://github.com/verida/verida-js/pull/243) which added the functionality to auto-create a `proof` string in the `verificationMethod` of the DID document for signing keys.

### PR DESCRIPTION
**Pull request by ITStar10**

### CODE REVIEW

File: VDA-Verification-Base/test/proof.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            assert.equal(proof, didDocumentContextProof)
+
+            const orgNonce = (await contract.getNonce(ADDRESS)).toNumber()
+            const params = ethers.utils.solidityPack(
+                ['string', 'uint'],
+                ['Test data', orgNonce]
+            )
+            const signature = await keyring.sign(params)
+
+            await contract.verifyRequestWithArray(ADDRESS, [ADDRESS], params, signature, didDocumentContextProof);
```
Comment (tahpot) [inline VDA-Verification-Base/test/proof.ts]: Should get the response and assert it is okay

File: VDA-Verification-Base/test/proof.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        contract = (await upgrades.deployProxy(
+            contractFactory,
+            {
+                initializer: 'initialize'
+            }
+        )) as TestContract
+        await contract.deployed()
+    })
+
+    describe('Keyring test', () => {
```
Comment (tahpot) [inline VDA-Verification-Base/test/proof.ts:51]: Can this test be removed now?

**Review by tahpot — COMMENTED** [review]
What if we merged into a single function that uses the provided `validSigners` if provided, otherwise it uses the signers stored in the contract?

```
EnumerableSetUpgradeable.AddressSet protected _trustedSigners

function verifyRequest(
        address did, 
        bytes memory params, 
        bytes calldata signature, 
        bytes calldata proof,
        address[] calldata validSigners,
    ) {
  
  index = 0
  length = 0
  memorySigner = false
  if (validSigners.length == 0) {
    memorySigner = true
    length = this._trustedSigners.length()
  } else {
    memorySigner = false
    length = validSigners.length
  }

  while (index < length && !isVerified) {
    if (memorySigner) {
        account = this._trustedSigners.at(index)
    } else {
        account = validSigners[index]
    }

    ... rest of the code
  }
```

I think it makes sense that this contract specifies the variable name for the white list of trusted signers (`_trustedSigners`). It should also support adding and removing them, so you could move that from the DID linkage branch.

Thoughts?

File: VDA-Verification-Base/test/proof.ts (hunk 1 above)
Comment (ITStar10) [reply VDA-Verification-Base/test/proof.ts]: See line number 180. It checks whether the nonce updated.

File: VDA-Verification-Base/test/proof.ts (hunk 2 above)
Comment (ITStar10) [reply VDA-Verification-Base/test/proof.ts:51]: I think, we can leave it for later reference.

**Review by tahpot — APPROVED** [review]

**Pull request merged**